### ISSUE TITLE
libqt5svg5 only appears as "Recomended Package" on installation

### ISSUE THREAD

**Issue by SamuAlfageme** [owncloud/client#6515]
I'm not sure of this (as we're seeing some more issues with the client's aspect with Qt 5.10), but It seems fishy since https://github.com/owncloud/client/commit/2bf1f27c80e73936027aae6eb8233b7f2d65568d happened really recently. 

I was installing the daily build over Ubuntu just now and noticed: 

```
sudo apt-get install owncloud-client
Reading package lists... Done
Building dependency tree
Reading state information... Done
The following additional packages will be installed:
  libowncloudsync0 ocqt5101-libqt5core5a ocqt5101-libqt5dbus5 ocqt5101-libqt5gui5 ocqt5101-libqt5keychain1 ocqt5101-libqt5network5 ocqt5101-libqt5printsupport5 ocqt5101-libqt5sql5
  ocqt5101-libqt5sql5-sqlite ocqt5101-libqt5webkit5 ocqt5101-libqt5widgets5 ocqt5101-libqt5xml5 ocqt5101-qtsvg5 owncloud-client-l10n
Suggested packages:
  ocqt5101-libqt5libqgtk2 owncloud-client-dolphin owncloud-client-nautilus owncloud-client-nemo owncloud-client-caja
Recommended packages:
  ocqt5101-libqt5svg5
The following NEW packages will be installed:
  libowncloudsync0 ocqt5101-libqt5core5a ocqt5101-libqt5dbus5 ocqt5101-libqt5gui5 ocqt5101-libqt5keychain1 ocqt5101-libqt5network5 ocqt5101-libqt5printsupport5 ocqt5101-libqt5sql5
  ocqt5101-libqt5sql5-sqlite ocqt5101-libqt5webkit5 ocqt5101-libqt5widgets5 ocqt5101-libqt5xml5 ocqt5101-qtsvg5 owncloud-client owncloud-client-l10n
```

See `Recommended packages:  ocqt5101-libqt5svg5` - won't be installed unless explicitly specified. 

I believe this causes the icon not to be displayed at all on:

<p align="center">
  <img src="https://user-images.githubusercontent.com/2644445/39920516-97acf768-5517-11e8-9316-035a995ef7f5.png"/>
</p>

**Comment by ogoffart** [comment]
Yes, it needs to be a dependence.

**Comment by guruz** [comment]
@jnweiger @michaelstingl FYI

**Comment by jnweiger** [comment]
The templates list @PACKAGE_NAME_PREFIX@-qtsvg5 always as a dependency, never as a recommends. hmmm.

**Comment by guruz** [comment]
@jnweiger So the newline is not a problem?
I think it's odd you let libowncloudsync0 be depended instead of owncloud-client (or are we using SVG in the sync library?!)

**Comment by jnweiger** [comment]
The control files referenced above show that  
ocqt5101-qtsvg5 is an unconditional install dependency of libowncloudsync0 and libowncloudsync0 is an unconditional install dependency of owncloud-client. I checked for stray newlines or linefeeds, could not find any. Please be more specific.

If you want me to move  ocqt5101-qtsvg5 to become a direct dependency of owncloud-client rather than an indirect one, please let me know.

The entire issue seems to be a non-issue to me:
with /etc/apt/apt.conf file:

    APT::Get::Install-Recommends "false";
    APT::Get::Install-Suggests "false";

and even with 

    apt-get --no-install-recommends install testpilotcloud-client

we always get ocqt5101-libqt5svg5 installed. It is a hard dependency, even if the "helpful" messages of apt install indicate otherwise. I think, apt is just too stupid, to skip `Recommends:` that are also hard dependencies.

Closing as worksforme. Please repopen if there is a reproducer, where ocqt5101-libqt5svg5 does not get installed.

**Comment by guruz** [comment]
meh same Icon missing on windows.
I'll try to fix NSIS/MSI

**Comment by guruz** [comment]
@dschmidt Can you port those 2 commits to MSVC+MSI?

**Comment by jnweiger** [comment]
This Ubuntu packaging issue now turns into a windows codig issue? -> I am outta here.
(unless @dschmidt needs my help)

**Comment by SamuAlfageme** [comment]
@dschmidt did you make it happen for msi? Asking 'cuz I can't reproduce with latest 2.6.0 daily build.

**Comment by guruz** [comment]
This is still happening on Linux, at least on Debian_8.0

```
00:05:15 Recommended packages:
00:05:15   libarchive-tar-perl ocqt5101-libqt5svg5
```
Debian 9:
```
00:08:52 Recommended packages:
00:08:52   ocqt5101-libqt5svg5
```

It is NOT showing up on other sections like ```The following extra packages will be installed:```

Since @jnweiger is on holiday maybe @crrodriguez can help?

**Comment by guruz** [comment]
(Also a good chance to integrate this test into Jenkins' client-linux-tests :-)

**Comment by dschmidt** [comment]
The recommendation is gone now.
The qtbase package `recommended` ocqt5101-libqt5svg5 which does not exist. Only ocqt5101-qtsvg5 exists and contains both the lib and the imageformat plugin. That's why I fixed the qtbase package to recommend this instead. The correct package was already required by the client (as @jnweiger correctly stated), thus nothing was broken in the first place, just a cosmetic issue.

```
docker run -ti debian:8.0                                                                                                                                                                                                   
root@14a19863f736:/# echo 'deb http://download.opensuse.org/repositories/isv:/ownCloud:/desktop:/daily:/master/Debian_8.0/ /' > /etc/apt/sources.list.d/isv:ownCloud:desktop:daily:master.list
root@14a19863f736:/# apt-get update


Ign http://download.opensuse.org  InRelease
Get:1 http://security.debian.org jessie/updates InRelease [44.9 kB]
Ign http://httpredir.debian.org jessie InRelease                                                                                                                                                                                                   
Get:2 http://httpredir.debian.org jessie-updates InRelease [145 kB]                                                                                                                                                                                
Get:3 http://download.opensuse.org  Release.gpg [477 B]                                                                                                                                                                                            
Get:4 http://download.opensuse.org  Release [1049 B]                                                                                                                                                                                               
Get:5 http://httpredir.debian.org jessie Release.gpg [2420 B]                                                                                                                                                                                      
Get:6 http://httpredir.debian.org jessie Release [148 kB]                                                                                                                                                                                          
Ign http://download.opensuse.org  Release                                                                                                                                                                                                          
Get:7 http://download.opensuse.org  Packages [13.4 kB]                                                                                                                                                                                             
Get:8 http://security.debian.org jessie/updates/main amd64 Packages [650 kB]                                                                                                                                                                       
Get:9 http://httpredir.debian.org jessie-updates/main amd64 Packages [23.0 kB]                                                                                                                                                                     
Get:10 http://httpredir.debian.org jessie/main amd64 Packages [9098 kB]                                                                                                                                                                            
Fetched 10.1 MB in 23s (423 kB/s)                                                                                                                                                                                                                  
Reading package lists... Done
W: GPG error: http://download.opensuse.org  Release: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 4ABE1AC7557BEFF9
root@14a19863f736:/# apt-get install owncloud-client
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following extra packages will be installed:
  dbus fontconfig fontconfig-config fonts-dejavu-core krb5-locales libalgorithm-c3-perl libarchive-extract-perl libavahi-client3 libavahi-common-data libavahi-common3 libbsd0 libcap-ng0 libcgi-fast-perl libcgi-pm-perl libclass-c3-perl libclass
  libdbus-1-3 libdrm-intel1 libdrm-nouveau2 libdrm-radeon1 libdrm2 libedit2 libelf1 libexpat1 libfcgi-perl libffi6 libfontconfig1 libfreetype6 libgdbm3 libgl1-mesa-dri libgl1-mesa-glx libglapi-mesa libglew1.10 libglib2.0-0 libglib2.0-data libg
  libgssapi-krb5-2 libhogweed2 libice6 libicu52 libjpeg62-turbo libk5crypto3 libkeyutils1 libkrb5-3 libkrb5support0 libllvm3.5 liblog-message-perl liblog-message-simple-perl libmodule-build-perl libmodule-pluggable-perl libmodule-signature-per
  libpackage-constants-perl libparams-util-perl libpciaccess0 libpng12-0 libpod-latex-perl libpod-readme-perl libregexp-common-perl libsecret-1-0 libsecret-common libsm6 libsoftware-license-perl libsqlite3-0 libssl1.0.0 libsub-exporter-perl li
  libtext-template-perl libtxc-dxtn-s2tc0 libx11-6 libx11-data libx11-xcb1 libxau6 libxcb-dri2-0 libxcb-dri3-0 libxcb-glx0 libxcb-icccm4 libxcb-image0 libxcb-keysyms1 libxcb-present0 libxcb-randr0 libxcb-render-util0 libxcb-render0 libxcb-shap
  libxcb-xkb1 libxcb1 libxdamage1 libxdmcp6 libxext6 libxfixes3 libxi6 libxkbcommon-x11-0 libxkbcommon0 libxml2 libxmu6 libxrender1 libxshmfence1 libxslt1.1 libxt6 libxxf86vm1 mesa-utils ocqt5101-libqt5concurrent5 ocqt5101-libqt5core5a ocqt510
  ocqt5101-libqt5network5 ocqt5101-libqt5printsupport5 ocqt5101-libqt5sql5 ocqt5101-libqt5sql5-sqlite ocqt5101-libqt5webkit5 ocqt5101-libqt5widgets5 ocqt5101-libqt5xml5 ocqt5101-qtsvg5 owncloud-client-l10n perl perl-base perl-modules rename sg
Suggested packages:
  dbus-x11 cups-common glew-utils gnome-keyring gnutls-bin krb5-doc krb5-user pciutils libthai0 ocqt5101-libqt5libqgtk2 owncloud-client-dolphin owncloud-client-nautilus owncloud-client-nemo owncloud-client-caja perl-doc libterm-readline-gnu-pe
  libcpanplus-perl libfile-checktree-perl libobject-accessor-perl sgml-base-doc debhelper
Recommended packages:
  libarchive-tar-perl
The following NEW packages will be installed:
  dbus fontconfig fontconfig-config fonts-dejavu-core krb5-locales libalgorithm-c3-perl libarchive-extract-perl libavahi-client3 libavahi-common-data libavahi-common3 libbsd0 libcap-ng0 libcgi-fast-perl libcgi-pm-perl libclass-c3-perl libclass
  libdbus-1-3 libdrm-intel1 libdrm-nouveau2 libdrm-radeon1 libdrm2 libedit2 libelf1 libexpat1 libfcgi-perl libffi6 libfontconfig1 libfreetype6 libgdbm3 libgl1-mesa-dri libgl1-mesa-glx libglapi-mesa libglew1.10 libglib2.0-0 libglib2.0-data libg
  libgssapi-krb5-2 libhogweed2 libice6 libicu52 libjpeg62-turbo libk5crypto3 libkeyutils1 libkrb5-3 libkrb5support0 libllvm3.5 liblog-message-perl liblog-message-simple-perl libmodule-build-perl libmodule-pluggable-perl libmodule-signature-per
  libpackage-constants-perl libparams-util-perl libpciaccess0 libpng12-0 libpod-latex-perl libpod-readme-perl libregexp-common-perl libsecret-1-0 libsecret-common libsm6 libsoftware-license-perl libsqlite3-0 libssl1.0.0 libsub-exporter-perl li
  libtext-template-perl libtxc-dxtn-s2tc0 libx11-6 libx11-data libx11-xcb1 libxau6 libxcb-dri2-0 libxcb-dri3-0 libxcb-glx0 libxcb-icccm4 libxcb-image0 libxcb-keysyms1 libxcb-present0 libxcb-randr0 libxcb-render-util0 libxcb-render0 libxcb-shap
  libxcb-xkb1 libxcb1 libxdamage1 libxdmcp6 libxext6 libxfixes3 libxi6 libxkbcommon-x11-0 libxkbcommon0 libxml2 libxmu6 libxrender1 libxshmfence1 libxslt1.1 libxt6 libxxf86vm1 mesa-utils ocqt5101-libqt5concurrent5 ocqt5101-libqt5core5a ocqt510
  ocqt5101-libqt5network5 ocqt5101-libqt5printsupport5 ocqt5101-libqt5sql5 ocqt5101-libqt5sql5-sqlite ocqt5101-libqt5webkit5 ocqt5101-libqt5widgets5 ocqt5101-libqt5xml5 ocqt5101-qtsvg5 owncloud-client owncloud-client-l10n perl perl-modules ren
The following packages will be upgraded:
  perl-base
1 upgraded, 146 newly installed, 0 to remove and 47 not upgraded.
Need to get 64.3 MB of archives.
After this operation, 239 MB of additional disk space will be used.
Do you want to continue? [Y/n] WARNING: The following packages cannot be authenticated!
  ocqt5101-libqt5core5a ocqt5101-libqt5concurrent5 ocqt5101-libqt5dbus5 ocqt5101-libqt5network5 ocqt5101-libqt5gui5 ocqt5101-libqt5keychain1 ocqt5101-libqt5widgets5 ocqt5101-qtsvg5 libowncloudsync0 ocqt5101-libqt5printsupport5 ocqt5101-libqt5s
  owncloud-client-l10n owncloud-client
Install these packages without verification? [y/N]
```

**Comment by dschmidt** [comment]
... and yes, the NSIS fix was also ported to MSI.
